src/common/RefCountedObj.h

```diff
@@ -27,6 +27,9 @@
 
 struct RefCountedObject {
 public:
+  using ref = boost::intrusive_ptr<RefCountedObject>;
```

Comment:
i'd suggest use free functions for these alias instead of embedding them as members.

Reply:
As an example, something like `type_ref<Message>`:

````
template<typename T>
using type_ref = boost::intrusive_ptr<T>;
````

Is this what you have in mind?

Reply:
yes, something like that. but i don't quite like the name `type_ref`, as "type" here is redundant. probably we can have an alias in `ceph` namespace, like
```c++
namespace ceph {
template<typename T>
using ref_t = boost::intrusive_ptr<T>;
}
template<typename T>
using cref_t = boost::intrusive_ptr<T>;
}
```
as the standard library already have `std::ref<>` and `std::cref<>`.

Reply:
Done, PTAL.